### ISSUE TITLE
Problem with list's key made of multiple leaf identifiers

### ISSUE THREAD

**Issue by asn-lns-sa-rj** [robshakir/pyangbind#290]
I encountered the problem with ietf-network-topology YANG model, but this is just an example to explain the issue with pyangbind. In this model, there are some lists that uses keys with one or more leaf identifiers. For example,  
- supporting-network [network-ref]
- supporting-node [network-ref node-ref]
- supporting-termination-point [network-ref node-ref tp-ref]
- supporting-link [network-ref link-ref]
I generated python classes for the model with pyangbind and started to populate some instances of network.
For example, a supporting network 'nw8' of an existing network 'nw2' can be easily specified this way 
>>> asn.N.networks.network['nw2'].supporting_network.add('nw8')
However I do not understand how I could do the same when the list's key is comprised of more than one leaf identifier.
Example 
>>> asn.N.networks.network['nw1'].node['node1.1'].supporting_node.add('nw2', 'node2.1')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Roland\AppData\Local\Programs\Python\Python310\lib\site-packages\pyangbind\lib\yangtypes.py", line 760, in add
    self.__set(_k=k)
  File "C:\Users\Roland\AppData\Local\Programs\Python\Python310\lib\site-packages\pyangbind\lib\yangtypes.py", line 624, in __set
    raise KeyError("YANGList key must contain all key elements (%s)"
**KeyError: "YANGList key must contain all key elements (['network_ref', 'node_ref'])"**

If we look at the generated classes below, we can compare definitions between supporting_network (single leaf key) and supporting_node (two-leaf key) 

class yc_supporting_network_submarine_networks__networks_network_supporting_network(PybindBase):
_pyangbind_elements = OrderedDict([('network_ref', network_ref), ])

class yc_supporting_node_submarine_networks__networks_network_node_supporting_node(PybindBase):
_pyangbind_elements = OrderedDict([('network_ref', network_ref), ('node_ref', node_ref), ])

Are keys with multiple leaf identifiers supported by pyang bind ? Or do I use a wrong syntax ?
Thanks for your help.

**Comment by xavier-contreras** [comment]
This should be supported, did you try the following by chance?

`asn.N.networks.network['nw1'].node['node1.1'].supporting_node.add(network_ref='nw2', node_ref='node2.1')`

**Comment by asn-lns-sa-rj** [comment]
Hi Xavier, 
Thanks for your reply. Indeed, I've found the right way to do that a few day later I posted my message, but I forgot to mention it on this thread.
I confirm, it is perfectly supported. 
In fact, the appropriate approach is simply to use the key exactly as it is in YANG. For example, in the ietf-network-topology case, supporting-termination-point ['1 1.1 1.1.1'] works in the case where network-ref = '1',  node-ref = '1.1', and tp-ref = '1.1.1']. Initially, I had not thought about that approach, just because it not pythonic !

Regards.
